system-linux.c

```diff
@@ -2362,11 +2366,35 @@ static const struct {
 #endif
 };
 
-static void system_add_link_modes(struct blob_buf *b, __u32 mask)
+static void system_add_link_modes_legacy(struct blob_buf *b, __u32 mask)
 {
-	size_t i;
+	uint32_t i;
```

Comment:
The `ARRAY_SIZE(...)` macro call yields a `size_t` type, why the change to `uint32_t`?